File: class/class4.1/src/Menu.cpp
#include "Menu.hpp"
#include <fstream>
#include <sstream>
#include <stdexcept>

Menu::Menu(const std::string& filePath) {
    loadIngredients(filePath);
}

void Menu::loadIngredients(const std::string& filePath) {
    std::ifstream file(filePath);
    if (!file) {
        throw std::runtime_error("Could not open file");
    }

    std::string line;
    while (std::getline(file, line)) {
        std::istringstream iss(line);
        std::string name;
        double cost;
        if (iss >> name >> cost) {
            ingredients.emplace_back(name, cost);
        }
    }
}

std::vector<Ingredient> Menu::getIngredients() const {
    return ingredients;
}

File: home/home4.1/pizza-builder/src/menu.rs
use crate::ingredient::Ingredient;
use std::fs::File;
use std::io::{BufRead, BufReader};

pub struct Menu {
    ingredients: Vec<Ingredient>,
}

impl Menu {
    pub fn new(file_path: &str) -> Self {
        let mut menu = Self {
            ingredients: Vec::new(),
        };
        menu.load_ingredients(file_path);
        menu
    }

    fn load_ingredients(&mut self, file_path: &str) {
        let file = File::open(file_path).expect("Could not open file");
        let reader = BufReader::new(file);

        for line in reader.lines() {
            let line = line.expect("Could not read line");
            let parts: Vec<&str> = line.split_whitespace().collect();
            if parts.len() == 2 {
                let name = parts[0];
                let cost = parts[1].parse::<f64>().expect("Could not parse cost");
                self.ingredients.push(Ingredient::new(name, cost));
            }
        }
    }

    pub fn get_ingredients(&self) -> &Vec<Ingredient> {
        &self.ingredients
    }
}
